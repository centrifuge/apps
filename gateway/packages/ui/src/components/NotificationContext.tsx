import React, { Component } from 'react';
import { Modal } from '@centrifuge/axis-modal';
import { Box, Button, Paragraph } from 'grommet';

export enum NOTIFICATION {
  DEFAULT = 'NOTIFICATION.DEFAULT',
  SUCCESS = 'NOTIFICATION.SUCCESS',
  WARNING = 'NOTIFICATION.WARNING',
  ERROR = 'NOTIFICATION.ERROR',
}


export type NotificationApi = {
  notify: (options: NotificationOptions) => void;
  alert: (options: AlertOptions) => void;
  close: () => void;
}

let CONTEXT_API: NotificationApi = {
  notify: (options: NotificationOptions) => {
  },
  alert: (options: AlertOptions) => {
  },
  close: () => {
  },

};

export const NotificationContext = React.createContext(CONTEXT_API);
export const NotificationConsumer = NotificationContext.Consumer;

export interface NotificationState {
  opened: boolean,
  options: NotificationOptions
}


export interface AlertOptions {
  title?: string,
  message?: string,
  cancelable?: boolean, // modal can self close(x icon and click outside)
  type?: NOTIFICATION, // DEFAULT, SUCCESS, ERROR, WARNING
  confirmLabel?: string, // Label for the modal button
  onClose?: () => void, // callback for when the modal is closed.
  onConfirm?: () => void, // callback for when the modal is closed.
}

export interface NotificationOptions {
  title?: string,
  message?: string,
  cancelable?: boolean, // modal can self close(x icon and click outside)
  type?: NOTIFICATION, // DEFAULT, SUCCESS, ERROR, WARNING
  confirmLabel?: string, // Label for the modal button
  onClose?: () => void, // callback for when the modal is closed.
  onConfirm?: () => void, // callback for when the modal is closed.
}


const DEFAULT_STATE: NotificationState = {
  opened: false,
  options: {
    title: '',
    message: '',
    cancelable: true,
    type: NOTIFICATION.DEFAULT,
    confirmLabel: 'OK',
  },
};

export class NotificationProvider extends Component<{}, NotificationState> {
  state = DEFAULT_STATE;

  constructor(props) {
    super(props);
    this.state = {
      ...DEFAULT_STATE,
    };

    CONTEXT_API = {
      notify: this.notify,
      alert: this.alert,
      close: this.close,
    };

  }


  alert = (options: AlertOptions) => {
    this.setState({
      opened: true,
      options: {
        ...this.state.options,
        ...options,
      },
    });
  };

  notify = (options: NotificationOptions) => {
    this.setState({
      opened: true,
      options: {
        ...this.state.options,
        ...options,
      },
    });
  };

  close = () => {
    this.state.options.onClose && this.state.options.onClose();
    this.setState(DEFAULT_STATE);
  };

  confirm = () => {
    this.state.options.onConfirm && this.state.options.onConfirm();
    this.close();
  };

  render() {
    const { children } = this.props;
    const { opened, options } = this.state;

    let modalProps: any = {
      headingProps: {
        level: 3,
        color: '',
      },
    };
    // Add close button to modal
    if (options.cancelable) {
      modalProps.onClose = this.close;
    }

    // default notification content.
    let modalContent = <Paragraph>
      {options.message}
    </Paragraph>;
    // default notification actions
    // We can extend to add more actions but we must be very careful at the keys
    let actions =  [
      <Button key={options.confirmLabel} primary label={options.confirmLabel} fill={false} onClick={this.confirm}/>
    ]

    // Compute the color of the modal title
    switch (options.type) {
      case NOTIFICATION.SUCCESS:
        modalProps.headingProps.color = 'status-ok';
        break;
      case NOTIFICATION.ERROR:
        modalProps.headingProps.color = 'status-error';
        break;
      case NOTIFICATION.WARNING:
        modalProps.headingProps.color = 'status-warning';
        break;

      default:
        actions = [];
        break
    }


    return (
      <NotificationContext.Provider
        value={CONTEXT_API}
      >
        <Modal
          opened={opened}
          title={options.title}
          {...modalProps}
        >
          <Box>
            {modalContent}
          </Box>

          <Box direction="row" gap="medium" justify={'end'}>
            {actions}
          </Box>
        </Modal>
        {children}
      </NotificationContext.Provider>
    );
  }
}

