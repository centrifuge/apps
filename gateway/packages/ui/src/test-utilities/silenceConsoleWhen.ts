type MessageFilter = string | RegExp | ((e: any) => boolean)

export const silenceConsoleWhen = (...filters: MessageFilter[]) => {
  const origConsoleError = console.error

  beforeAll(() => {
    console.error = (e) => {
      const msg = e && e.message ? e.message : typeof e === 'string' ? e : ''
      for (let i = 0; i < filters.length; i += 1) {
        const filter = filters[i]

        if (
          (typeof filter === 'string' && msg.indexOf(filter) !== -1) ||
          (filter instanceof RegExp && msg.match(filter)) ||
          (typeof filter === 'function' && filter(e))
        ) {
          return
        }
      }
      origConsoleError(e)
    }
  })

  afterAll(() => {
    console.error = origConsoleError
  })
}
