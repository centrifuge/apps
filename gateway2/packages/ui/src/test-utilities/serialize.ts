import toJson from 'enzyme-to-json';
// TODO we shoud remove this
export const serializeSnapshot = (component) => {
  // We are not interested in the theme prop and we remove it in order to reduce file size
  return toJson(
    component,
    {
      noKey: false,
      mode: 'deep',
      map: (node => {
        //if(node.type.displayName.includes('WithTheme'))
        delete node.props.theme;
        if(node.props.dropContent) {
          node.props.dropContent = null
        }

        return node;
      }),
    },
  );
};
