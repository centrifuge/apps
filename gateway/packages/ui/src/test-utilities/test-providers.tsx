import { AxisTheme } from '@centrifuge/axis-theme'
import { User } from '@centrifuge/gateway-lib/models/user'
import React from 'react'
import { AppContext } from '../App'
import { NotificationProvider } from '../components/NotificationContext'
import { defaultUser } from './default-data'

// Wrap component with Application Context;
export const withApplicationContext = (Component, user: User = defaultUser, setUser = () => {}) => {
  return <AppContext.Provider value={{ user, setUser }}>{Component}</AppContext.Provider>
}

// Wrap component with Notification Context;
export const withNotificationContext = (Component) => {
  return <NotificationProvider>{Component}</NotificationProvider>
}

// Wrap in Axis Theme
export const withAxis = (Component) => {
  return <AxisTheme full={true}>{Component}</AxisTheme>
}

export const withAllProvidersAndContexts = (Component, user: User = defaultUser, setUser = () => {}) => {
  return withAxis(withApplicationContext(withNotificationContext(Component), user, setUser))
}
