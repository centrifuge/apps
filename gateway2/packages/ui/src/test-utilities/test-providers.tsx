import { User } from '@centrifuge/gateway-lib/models/user';
import React from 'react';
import { AuthContext } from '../auth/Auth';
import { NotificationProvider } from '../components/NotificationContext';
import { AxisTheme } from '@centrifuge/axis-theme';
import { defaultToken, defaultUser } from './default-data';

// Wrap component with Application Context;
export const withApplicationContext = (
  Component,
  user: User = defaultUser,
  setUser = () => {},
  token = defaultToken,
  setToken = () => {},
) => {
  return (
    <AuthContext.Provider value={{ user, setUser, token, setToken }}>
      {Component}
    </AuthContext.Provider>
  );
};

// Wrap component with Notification Context;
export const withNotificationContext = Component => {
  return <NotificationProvider>{Component}</NotificationProvider>;
};

// Wrap in Axis Theme
export const withAxis = Component => {
  return <AxisTheme full={true}>{Component}</AxisTheme>;
};

export const withAllProvidersAndContexts = (
  Component,
  user: User = defaultUser,
  setUser = () => {},
  token = defaultToken,
  setToken = () => {},
) => {
  return withAxis(
    withApplicationContext(
      withNotificationContext(Component),
      user,
      setUser,
      token,
      setToken,
    ),
  );
};
