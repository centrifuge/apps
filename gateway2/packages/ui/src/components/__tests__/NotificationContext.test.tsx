import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { AxisTheme } from '@centrifuge/axis-theme';
import { NOTIFICATION, NotificationConsumer, NotificationProvider } from '../NotificationContext';
import { mount, shallow } from 'enzyme';
import { Modal } from '@centrifuge/axis-modal';
import { serializeSnapshot } from '../../test-utilities/serialize';
import { withAxis, withNotificationContext } from '../../test-utilities/test-providers';


const setProviders = (component) => {
  return withAxis(
    withNotificationContext(
      component,
    ),
  );
};

describe('NotificationContext', () => {

  it('Should not crash', () => {
    const tree = shallow(
      setProviders(
        <NotificationConsumer>
          {({ notify }) => {
            return <p>Test render</p>;
          }}
        </NotificationConsumer>,
      ),
    );
    expect(serializeSnapshot(tree)).toMatchSnapshot();
  });


  it('Should render a default popup', () => {

    const tree = mount(
      setProviders(
        <NotificationConsumer>
          {({ notify }) => {
            return <button onClick={() => {
              notify(
                {
                  title: 'Title',
                  message: 'Messsge',
                },
              );
            }}></button>;
          }}
        </NotificationConsumer>,
      ),
    );
    tree.find('button').simulate('click');
    expect(serializeSnapshot(tree.find(Modal))).toMatchSnapshot();
  });

  it('Should render a warning popup', () => {
    const tree = mount(
      setProviders(
        <NotificationConsumer>
          {({ notify }) => {
            return <button onClick={() => {
              notify(
                {
                  title: 'Title',
                  message: 'Messsge',
                  type: NOTIFICATION.WARNING,
                },
              );
            }}></button>;
          }}
        </NotificationConsumer>,
      ),
    );
    tree.find('button').simulate('click');
    expect(serializeSnapshot(tree.find(Modal))).toMatchSnapshot();
  });


  it('Should render a success popup', () => {
    const tree = mount(
      setProviders(
        <NotificationConsumer>
          {({ notify }) => {
            return <button onClick={() => {
              notify(
                {
                  title: 'Title',
                  message: 'Messsge',
                  type: NOTIFICATION.SUCCESS,
                },
              );
            }}></button>;
          }}
        </NotificationConsumer>,
      ),
    );
    tree.find('button').simulate('click');
    expect(serializeSnapshot(tree.find(Modal))).toMatchSnapshot();
  });

  it('Should render a error popup', () => {
    const tree = mount(
      setProviders(
        <NotificationConsumer>
          {({ notify }) => {
            return <button onClick={() => {
              notify(
                {
                  title: 'Title',
                  message: 'Messsge',
                  type: NOTIFICATION.ERROR,
                },
              );
            }}></button>;
          }}
        </NotificationConsumer>,
      ),
    );
    tree.find('button').simulate('click');
    expect(serializeSnapshot(tree.find(Modal))).toMatchSnapshot();
  });

  it('Should set the proper label for the confirm button', () => {
    const tree = mount(
      <AxisTheme>
        <NotificationProvider>
          <NotificationConsumer>
            {({ notify }) => {

              return <button onClick={() => {
                notify(
                  {
                    title: 'Title',
                    message: 'Messsge',
                    confirmLabel: 'Crazy Label',
                  },
                );
              }}></button>;
            }}
          </NotificationConsumer>
        </NotificationProvider>
      </AxisTheme>
      ,
    );

    tree.find('button').simulate('click');
    expect(serializeSnapshot(tree.find(Modal))).toMatchSnapshot();
  });

});
