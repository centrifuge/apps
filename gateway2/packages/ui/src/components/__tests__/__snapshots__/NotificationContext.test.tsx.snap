// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationContext Should not crash 1`] = `
<Grommet
  full={true}
>
  <GlobalStyleComponent />
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</Grommet>
`;

exports[`NotificationContext Should render a default popup 1`] = `
<Portal
  containerInfo={
    <div>
      <div
        aria-hidden="false"
      >
        <div
          class="StyledLayer-rmtehz-0 cfjZVQ"
          tabindex="-1"
        >
          <div
            class="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
          />
          <div
            class="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
          >
            <a
              aria-hidden="true"
              class="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
              tabindex="-1"
            />
            <div
              class="StyledBox-sc-13pk1d4-0 bvFSOw"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fVYJBt"
              >
                <h3
                  class="StyledHeading-sc-1rdh4aw-0 cutvkF"
                >
                  <div
                    class="Modal__Title-sc-14fi5dq-1 hTYitU"
                  >
                    Title
                  </div>
                </h3>
                <button
                  class="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
                  type="button"
                >
                  <svg
                    aria-label="Close"
                    class="StyledIcon-ofa7kd-0 cgOtHq"
                    style="width: 16px; height: 16px; margin-right: -20px; margin-left: -20px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3,3 L21,21 M3,21 L21,3"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 faQrIe"
              >
                <p
                  class="StyledParagraph-tbetod-0 dIDtVY"
                >
                  Messsge
                </p>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 cvDtre"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  }
>
  <div
    aria-hidden={false}
  >
    <div
      className="StyledLayer-rmtehz-0 cfjZVQ"
      onKeyDown={[Function]}
      tabIndex="-1"
    >
      <div
        className="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
        onMouseDown={[Function]}
      />
      <div
        className="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
      >
        <a
          aria-hidden="true"
          className="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
          tabIndex="-1"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 bvFSOw"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 fVYJBt"
          >
            <h3
              className="StyledHeading-sc-1rdh4aw-0 cutvkF"
            >
              <div
                className="Modal__Title-sc-14fi5dq-1 hTYitU"
              >
                Title
              </div>
            </h3>
            <button
              className="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-label="Close"
                className="StyledIcon-ofa7kd-0 cgOtHq"
                style={
                  Object {
                    "height": "16px",
                    "marginLeft": "-20px",
                    "marginRight": "-20px",
                    "width": "16px",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M3,3 L21,21 M3,21 L21,3"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </button>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 faQrIe"
          >
            <p
              className="StyledParagraph-tbetod-0 dIDtVY"
            >
              Messsge
            </p>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 cvDtre"
          />
        </div>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`NotificationContext Should render a error popup 1`] = `
<Portal
  containerInfo={
    <div>
      <div
        aria-hidden="false"
      >
        <div
          class="StyledLayer-rmtehz-0 cfjZVQ"
          tabindex="-1"
        >
          <div
            class="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
          />
          <div
            class="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
          >
            <a
              aria-hidden="true"
              class="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
              tabindex="-1"
            />
            <div
              class="StyledBox-sc-13pk1d4-0 bvFSOw"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fVYJBt"
              >
                <h3
                  class="StyledHeading-sc-1rdh4aw-0 knbKGX"
                >
                  <div
                    class="Modal__Title-sc-14fi5dq-1 hTYitU"
                  >
                    Title
                  </div>
                </h3>
                <button
                  class="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
                  type="button"
                >
                  <svg
                    aria-label="Close"
                    class="StyledIcon-ofa7kd-0 cgOtHq"
                    style="width: 16px; height: 16px; margin-right: -20px; margin-left: -20px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3,3 L21,21 M3,21 L21,3"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 faQrIe"
              >
                <p
                  class="StyledParagraph-tbetod-0 dIDtVY"
                >
                  Messsge
                </p>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 cvDtre"
              >
                <button
                  class="StyledButton-sc-323bzc-0 flUQlI"
                  type="button"
                >
                  OK
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  }
>
  <div
    aria-hidden={false}
  >
    <div
      className="StyledLayer-rmtehz-0 cfjZVQ"
      onKeyDown={[Function]}
      tabIndex="-1"
    >
      <div
        className="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
        onMouseDown={[Function]}
      />
      <div
        className="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
      >
        <a
          aria-hidden="true"
          className="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
          tabIndex="-1"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 bvFSOw"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 fVYJBt"
          >
            <h3
              className="StyledHeading-sc-1rdh4aw-0 knbKGX"
            >
              <div
                className="Modal__Title-sc-14fi5dq-1 hTYitU"
              >
                Title
              </div>
            </h3>
            <button
              className="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-label="Close"
                className="StyledIcon-ofa7kd-0 cgOtHq"
                style={
                  Object {
                    "height": "16px",
                    "marginLeft": "-20px",
                    "marginRight": "-20px",
                    "width": "16px",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M3,3 L21,21 M3,21 L21,3"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </button>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 faQrIe"
          >
            <p
              className="StyledParagraph-tbetod-0 dIDtVY"
            >
              Messsge
            </p>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 cvDtre"
          >
            <button
              className="StyledButton-sc-323bzc-0 flUQlI"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              OK
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`NotificationContext Should render a success popup 1`] = `
<Portal
  containerInfo={
    <div>
      <div
        aria-hidden="false"
      >
        <div
          class="StyledLayer-rmtehz-0 cfjZVQ"
          tabindex="-1"
        >
          <div
            class="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
          />
          <div
            class="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
          >
            <a
              aria-hidden="true"
              class="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
              tabindex="-1"
            />
            <div
              class="StyledBox-sc-13pk1d4-0 bvFSOw"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fVYJBt"
              >
                <h3
                  class="StyledHeading-sc-1rdh4aw-0 jXJQdf"
                >
                  <div
                    class="Modal__Title-sc-14fi5dq-1 hTYitU"
                  >
                    Title
                  </div>
                </h3>
                <button
                  class="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
                  type="button"
                >
                  <svg
                    aria-label="Close"
                    class="StyledIcon-ofa7kd-0 cgOtHq"
                    style="width: 16px; height: 16px; margin-right: -20px; margin-left: -20px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3,3 L21,21 M3,21 L21,3"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 faQrIe"
              >
                <p
                  class="StyledParagraph-tbetod-0 dIDtVY"
                >
                  Messsge
                </p>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 cvDtre"
              >
                <button
                  class="StyledButton-sc-323bzc-0 flUQlI"
                  type="button"
                >
                  OK
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  }
>
  <div
    aria-hidden={false}
  >
    <div
      className="StyledLayer-rmtehz-0 cfjZVQ"
      onKeyDown={[Function]}
      tabIndex="-1"
    >
      <div
        className="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
        onMouseDown={[Function]}
      />
      <div
        className="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
      >
        <a
          aria-hidden="true"
          className="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
          tabIndex="-1"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 bvFSOw"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 fVYJBt"
          >
            <h3
              className="StyledHeading-sc-1rdh4aw-0 jXJQdf"
            >
              <div
                className="Modal__Title-sc-14fi5dq-1 hTYitU"
              >
                Title
              </div>
            </h3>
            <button
              className="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-label="Close"
                className="StyledIcon-ofa7kd-0 cgOtHq"
                style={
                  Object {
                    "height": "16px",
                    "marginLeft": "-20px",
                    "marginRight": "-20px",
                    "width": "16px",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M3,3 L21,21 M3,21 L21,3"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </button>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 faQrIe"
          >
            <p
              className="StyledParagraph-tbetod-0 dIDtVY"
            >
              Messsge
            </p>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 cvDtre"
          >
            <button
              className="StyledButton-sc-323bzc-0 flUQlI"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              OK
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`NotificationContext Should render a warning popup 1`] = `
<Portal
  containerInfo={
    <div>
      <div
        aria-hidden="false"
      >
        <div
          class="StyledLayer-rmtehz-0 cfjZVQ"
          tabindex="-1"
        >
          <div
            class="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
          />
          <div
            class="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
          >
            <a
              aria-hidden="true"
              class="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
              tabindex="-1"
            />
            <div
              class="StyledBox-sc-13pk1d4-0 bvFSOw"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fVYJBt"
              >
                <h3
                  class="StyledHeading-sc-1rdh4aw-0 cTVBxu"
                >
                  <div
                    class="Modal__Title-sc-14fi5dq-1 hTYitU"
                  >
                    Title
                  </div>
                </h3>
                <button
                  class="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
                  type="button"
                >
                  <svg
                    aria-label="Close"
                    class="StyledIcon-ofa7kd-0 cgOtHq"
                    style="width: 16px; height: 16px; margin-right: -20px; margin-left: -20px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3,3 L21,21 M3,21 L21,3"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 faQrIe"
              >
                <p
                  class="StyledParagraph-tbetod-0 dIDtVY"
                >
                  Messsge
                </p>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 cvDtre"
              >
                <button
                  class="StyledButton-sc-323bzc-0 flUQlI"
                  type="button"
                >
                  OK
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  }
>
  <div
    aria-hidden={false}
  >
    <div
      className="StyledLayer-rmtehz-0 cfjZVQ"
      onKeyDown={[Function]}
      tabIndex="-1"
    >
      <div
        className="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
        onMouseDown={[Function]}
      />
      <div
        className="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
      >
        <a
          aria-hidden="true"
          className="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
          tabIndex="-1"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 bvFSOw"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 fVYJBt"
          >
            <h3
              className="StyledHeading-sc-1rdh4aw-0 cTVBxu"
            >
              <div
                className="Modal__Title-sc-14fi5dq-1 hTYitU"
              >
                Title
              </div>
            </h3>
            <button
              className="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-label="Close"
                className="StyledIcon-ofa7kd-0 cgOtHq"
                style={
                  Object {
                    "height": "16px",
                    "marginLeft": "-20px",
                    "marginRight": "-20px",
                    "width": "16px",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M3,3 L21,21 M3,21 L21,3"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </button>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 faQrIe"
          >
            <p
              className="StyledParagraph-tbetod-0 dIDtVY"
            >
              Messsge
            </p>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 cvDtre"
          >
            <button
              className="StyledButton-sc-323bzc-0 flUQlI"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              OK
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`NotificationContext Should set the proper label for the confirm button 1`] = `
<Portal
  containerInfo={
    <div>
      <div
        aria-hidden="false"
      >
        <div
          class="StyledLayer-rmtehz-0 cfjZVQ"
          tabindex="-1"
        >
          <div
            class="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
          />
          <div
            class="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
          >
            <a
              aria-hidden="true"
              class="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
              tabindex="-1"
            />
            <div
              class="StyledBox-sc-13pk1d4-0 bvFSOw"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 fVYJBt"
              >
                <h3
                  class="StyledHeading-sc-1rdh4aw-0 cutvkF"
                >
                  <div
                    class="Modal__Title-sc-14fi5dq-1 hTYitU"
                  >
                    Title
                  </div>
                </h3>
                <button
                  class="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
                  type="button"
                >
                  <svg
                    aria-label="Close"
                    class="StyledIcon-ofa7kd-0 cgOtHq"
                    style="width: 16px; height: 16px; margin-right: -20px; margin-left: -20px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3,3 L21,21 M3,21 L21,3"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 faQrIe"
              >
                <p
                  class="StyledParagraph-tbetod-0 dIDtVY"
                >
                  Messsge
                </p>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 cvDtre"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  }
>
  <div
    aria-hidden={false}
  >
    <div
      className="StyledLayer-rmtehz-0 cfjZVQ"
      onKeyDown={[Function]}
      tabIndex="-1"
    >
      <div
        className="StyledLayer__StyledOverlay-rmtehz-1 bYEbar"
        onMouseDown={[Function]}
      />
      <div
        className="StyledLayer__StyledContainer-rmtehz-2 ijiFTe"
      >
        <a
          aria-hidden="true"
          className="LayerContainer__HiddenAnchor-sc-1srj14c-0 LMZHJ"
          tabIndex="-1"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 bvFSOw"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 fVYJBt"
          >
            <h3
              className="StyledHeading-sc-1rdh4aw-0 cutvkF"
            >
              <div
                className="Modal__Title-sc-14fi5dq-1 hTYitU"
              >
                Title
              </div>
            </h3>
            <button
              className="Modal__CloseButton-sc-14fi5dq-2 bdvYji StyledButton-sc-323bzc-0 dmNIDB"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <svg
                aria-label="Close"
                className="StyledIcon-ofa7kd-0 cgOtHq"
                style={
                  Object {
                    "height": "16px",
                    "marginLeft": "-20px",
                    "marginRight": "-20px",
                    "width": "16px",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M3,3 L21,21 M3,21 L21,3"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </button>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 faQrIe"
          >
            <p
              className="StyledParagraph-tbetod-0 dIDtVY"
            >
              Messsge
            </p>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 cvDtre"
          />
        </div>
      </div>
    </div>
  </div>
</Portal>
`;
